### ISSUE TITLE
ue5-main: Internal error: Cannot specify -o when generating multiple output files

### ISSUE THREAD

**Issue by negril** [octobuild/octobuild#92]
When building current ue5-main octobuild complains about specifying multiple output files because it wrongly interprets response files as output files.

I added a debug line like this:

```
diff --git a/src/clang/prepare.rs b/src/clang/prepare.rs
index eb5f5ee..cbefc27 100644
--- a/src/clang/prepare.rs
+++ b/src/clang/prepare.rs
@@ -78,6 +78,7 @@ pub fn create_tasks(
         ParamValue::None => None,
         ParamValue::Single(v) => {
             if input_sources.len() > 1 {
+                println!("files: {:?}", input_sources);
                 return Err(crate::Error::from(
                     "Cannot specify -o when generating multiple output files",
```
resulting in 
`files: ["@../Intermediate/Build/Linux/x64/CrashReportClient/Development/AnalyticsET/AnalyticsET.Shared.rsp", "../Intermediate/Build/Linux/x64/CrashReportClient/Development/AnalyticsET/Module.AnalyticsET.cpp"]`

My fix would be to parse the arguments for response files and include them to the argument list. Note that the response files can include other response files. So you have to do the same pass of them as well.

The `.Shared.rsp` files often repeat compiler options already in the top `.rsp` file. I am not sure how much of an issue this is for the way you parse certain configs. But ideally this should by fixed upstream. Just something to be aware of.

**Comment by slonopotamus** [comment]
octobuild already handles response files, see [expand_response_files](https://github.com/octobuild/octobuild/blob/master/src/utils.rs#L18-L51) function.

> Note that the response files can include other response files.

This contradicts with the [documentation](https://learn.microsoft.com/en-us/cpp/build/reference/at-specify-a-compiler-response-file?view=msvc-170) that explicitly says:

> It is not possible to specify the @ option from within a response file. That is, a response file cannot embed another response file.

**Comment by slonopotamus** [comment]
Oh,, wait, you're talking about clang...

**Comment by slonopotamus** [comment]
For the reference, here's where these changes were introduced to UE: https://github.com/EpicGames/UnrealEngine/commit/0de807f1a8c8ec496fa39a1c7d070275c92a2643

**Comment by negril** [comment]
If I can help in anyway ping me.

**Comment by slonopotamus** [comment]
Fixed in 0.6.0.
